content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md

```diff
@@ -0,0 +1,382 @@
+service/elasticsearch-master-headless   ClusterIP   None             <none>        9200/TCP,9300/TCP     8m5s
+service/skywalking-oap                  ClusterIP   10.108.29.84     <none>        11800/TCP,12800/TCP   8m5s
+service/skywalking-ui                   ClusterIP   10.102.186.131   <none>        80/TCP                8m5s
+
+NAME                             READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/skywalking-oap   2/2     2            2           8m6s
+deployment.apps/skywalking-ui    1/1     1            1           8m6s
+
+NAME                                        DESIRED   CURRENT   READY   AGE
+replicaset.apps/skywalking-oap-7f757c7668   2         2         2       8m9s
+replicaset.apps/skywalking-ui-549dc5989f    1         1         1       8m9s
+
+NAME                                    READY   AGE
+statefulset.apps/elasticsearch-master   3/3     8m5s
+
+NAME                           COMPLETIONS   DURATION   AGE
+job.batch/skywalking-es-init   1/1           7m27s      8m6s
+```
+
+You can edit the deployed SkyWalking UI service *skywalking_ui* and change its type from *ClusterIP* to *NodePort*. The service will be automatically mapped to gateway host with an assigned port.
+
+```markdown
+$ k edit service/skywalking-ui -n skywalking
+
+$ k describe service/skywalking-ui -n skywalking 
+Name:                     skywalking-ui
+Namespace:                skywalking
+Labels:                   app=skywalking
+                          app.kubernetes.io/managed-by=Helm
+                          chart=skywalking-4.2.0
+                          component=ui
+                          heritage=Helm
+                          hpecp.hpe.com/hpecp-internal-gateway=true
+                          release=skywalking
+Annotations:              hpecp-internal-gateway/80: gl2-caas.gl-hpe.local:10037
+                          meta.helm.sh/release-name: skywalking
+                          meta.helm.sh/release-namespace: skywalking
+Selector:                 app=skywalking,component=ui,release=skywalking
+Type:                     NodePort
+IP:                       10.102.186.131
+Port:                     <unset>  80/TCP
+TargetPort:               8080/TCP
+NodePort:                 <unset>  32748/TCP
+Endpoints:                10.192.7.25:8080
+Session Affinity:         None
+External Traffic Policy:  Cluster
+Events:                   <none>
+```
+
+As shown in the ***Annotations*** section of the service description above, the SkyWalking UI can then be accessed in the browser by typing the address *gl2-caas.gl-hpe.local:10037*: 
+
+![](/img/sw-ui.png)
+
+### Deploy a sample application: *SpringBoot*
+
+A﻿s the first demo application, create a *SpingBoot* Web app that provides a REST endpoint **/message** to print some nice message. Then generate a *jar* package using the [Apache Maven](https://maven.apache.org/what-is-maven.html) command *mvn* with the *pom.xml* file:
```

Comment:
Do you really mean "As the first demo application..." or "As **in** the first demo application..." If it's the second, you should rephrase it to "As I had you do in the application I demo'd in my first blog post in this series, create...."

Add a comma after "Then" at the beginning of the next sentence.